streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -223,10 +223,7 @@ boolean handleCorruption(final Set<TaskId> corruptedTasks) {
             final Collection<Task> tasksToCommit = allTasks()
                 .values()
                 .stream()
-                // TODO: once we remove state restoration from the stream thread, we can also remove
-                //  the RESTORING state here, since there will not be any restoring tasks managed
-                //  by the stream thread anymore.
-                .filter(t -> t.state() == Task.State.RUNNING || t.state() == Task.State.RESTORING)
+                .filter(t -> t.state() == Task.State.RUNNING)
```

Comment:
Given that we still have a feature flag, should we make this condition more complex and consider if the state updater thread is enabled or not, and check different conditions for both cases?

Reply:
I do not think that this is needed. Don't you agree that restoring active tasks do not need to be committed -- with or without state updater.

Reply:
I was just reading the TODO without much thinking about it... -- I guess we might still want to flush restoring tasks and write the checkpoint file (what is part to a commit) -- so should we execute `preCommit()` and `postCommit()` for those -- I agree that we won't have input topic offsets to be committed (and the should not be any TX).

Reply:
Since restoring active tasks return false from `commitNeeded()` because they have never processed records and have never execute a punctuation, `preCommit()` and `postCommit()` are [never called on restoring active task in this specific code](https://github.com/apache/kafka/blob/c32d2338a7e0079e539b74eb16f0095380a1ce85/streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutor.java#L141). This is true for enabled and disabled state updater.
Additionally, as far as I know there is nothing to flush in a restoring active task, because restoration uses the state restore callback.

Reply:
Hey @cadonna sorry I came late on this PR.

One thing I'd like to raise is that in the past, we've seen active task restoring never complete under rolling restart / rebalance storm scenarios since we kept losing the progress we made thus far when reviving. I'm not 100% sure if this part of the code is related to that scenario but just try to double check. If you have thought about it and concluded this would not be related, I'm relieved :)

Reply:
@guozhangwang as far as I can see in the code, a restoring active task does not return `true` from `commitNeeded()`. Thus, `postCommit()` is never called here. Do you agree?